src/node_file.cc

```diff
@@ -3127,6 +3127,44 @@ static void GetFormatOfExtensionlessFile(
   return args.GetReturnValue().Set(EXTENSIONLESS_FORMAT_JAVASCRIPT);
 }
 
+#ifdef _WIN32
```

Comment:
libuv exports a number of similar functions. Before adding this to Node, do [any of these](https://github.com/libuv/libuv/blob/be0b00a80db6cb19da80850127c133e40597f5b0/include/uv.h#L1935-L1944) work for your use case?

Reply:
I don’t know, and I don’t have a Windows machine to try them out :/

Reply:
I can help verify. I didn't find any unicode utils in node, so was a bit surprised actually. Now it makes sense since it's from libuv.

Reply:
I played around with it. There are some boilerplates for that to work. One example how it can be used by node is this  [libuv's util function](https://github.com/libuv/libuv/blob/be0b00a80db6cb19da80850127c133e40597f5b0/src/win/util.c#L1039-L1060), which is, unfortunately, not exposed. Node would have to implement that or something similar, and pass in `c_str` pointers when calling it. 

What we have here, IMHO, is actually more suitable for node as it's operates directly on `std::string`

We can certainly take this out to a util file, e.g. `util.cc`, to make it more proper.

Reply:
Does it actually make sense to move it to `util.cc`? Given that it's path-only, `node_file.cc` doesn't seem out of place.

Reply:
Makes sense. I was thinking about like libuv to have a windows specific file to place these, namely util_win.cc but that would be a different PR. I think it's fine to have these here for now.